src/apollo.ts

```diff
@@ -11,6 +16,18 @@ const httpLink = new HttpLink({
 export const client = new ApolloClient({
   cache: new InMemoryCache({
     possibleTypes: generatedIntrospection.possibleTypes,
+    // we need to explicitly handle the uniqueness of chain object because
+    // ID is not unique across as different network can have the same ID
+    // which confuses the caching of apollo client.
+    // the code below is to override the handling of uniqueness for Chain type.
+    dataIdFromObject(responseObject) {
+      switch (responseObject.__typename) {
+        case 'Chain':
+          return `Chain:${responseObject.network}:${responseObject.id}`
```

Comment:
and network is guaranteed to be in the response?

Reply:
Yes, `network` in non nullable in the GQL schema  https://github.com/smartcontractkit/chainlink/blob/83a36e9c58ddc924d79adda3c8327a47c1231d1f/core/web/schema/type/chain.graphql#L5

Reply:
Not specifying the network in one of the fragments could potentially introduce a silent bug I think.

Reply:
@ChrisAmora  correct, that was the best quick fix i could do, the proper fix would be to change the backend API to return RelayID instead of chain ID as the graphql ID. But that is a breaking change and bigger fix.

What about throwing an error, so it fails loudly instead of silently for devs when they forget to load network?

<img width="865" alt="Screenshot 2024-12-09 at 12 29 30 pm" src="https://github.com/user-attachments/assets/d7757efc-5b7f-468f-b872-aa84653059a7">

Reply:
Yeah I think it's a good idea.